### PR TITLE
Feat/seed msig work

### PR DESCRIPTION
**Pull request by whyrusleeping**
includes #2305 and #2440

### CODE REVIEW

File: chain/gen/genesis/genesis.go [hunk 1]
```diff
@@ -195,33 +197,66 @@ func MakeInitialStateTree(ctx context.Context, bs bstore.Blockstore, template ge
 
 	// Create accounts
 	for id, info := range template.Accounts {
-		if info.Type != genesis.TAccount {
-			return nil, xerrors.New("unsupported account type") // TODO: msigs
+		if info.Type != genesis.TAccount && info.Type != genesis.TMultisig {
+			return nil, xerrors.New("unsupported account type")
```
Comment (travisperson) [inline chain/gen/genesis/genesis.go:201]: This should be an `||` I'm pretty sure.

File: chain/gen/genesis/t01_init.go [hunk 2]
```diff
@@ -30,6 +32,16 @@ func SetupInitActor(bs bstore.Blockstore, netname string, initialActors []genesi
 	amap := hamt.NewNode(cst, hamt.UseTreeBitWidth(5)) // TODO: use spec adt map
 
 	for i, a := range initialActors {
+		if a.Type == genesis.TMultisig {
+			addr, _ := address.NewActorAddress(a.Meta)
+			fmt.Printf("init set %s t0%d\n", addr, AccountStart+uint64(i))
+
+			if err := amap.Set(context.TODO(), string(addr.Bytes()), AccountStart+uint64(i)); err != nil {
+				return nil, err
+			}
```
Comment (magik6k) [inline chain/gen/genesis/t01_init.go:41]: That's not needed, is it?

File: cmd/lotus-seed/genesis.go [hunk 3]
```diff
@@ -141,3 +149,153 @@ var genesisAddMinerCmd = &cli.Command{
...
+			Version:       1,
+			ID:            e[1],
+			Amount:        balance,
+			CustodianID:   custodianID,
+			VestingMonths: vesting,
+			M:             threshold,
+			N:             num,
+			Type:          e[8],
+			Sig1:          e[9],
+			Sig2:          e[10],
```
Comment (magik6k) [inline cmd/lotus-seed/genesis.go:295]: Those seem to be unused, is this intentional?

Comment (whyrusleeping) [reply cmd/lotus-seed/genesis.go:295]: They will be used later, for now its just there to keep the input format consistent

File: chain/gen/genesis/t01_init.go (hunk 2 above)
Comment (whyrusleeping) [reply chain/gen/genesis/t01_init.go:41]: mmmm, no. i think we can get away with just an ID address for all of these

**Pull request merged**